### PR TITLE
Improve tests for "swift" tasks

### PR DESCRIPTION
**Pull request by award999**
* Move what can be to the unit level
* Add in more unit test cases to cover additional branches
* Add in integration tests where needed to increase coverage

Issue: #1038

### CODE REVIEW

**Comment by award999** [comment]
@swift-server-bot test this please

**Comment by award999** [comment]
@swift-server-bot test this please

**Comment by award999** [comment]
@swift-server-bot test this please

**Comment by award999** [comment]
@swift-server-bot test this please

File: assets/test/.vscode/tasks.json [hunk 1]
```diff
@@ -20,15 +20,15 @@
...
+				"--show-bin-path"
 			],
 			"cwd": "defaultPackage",
 			"problemMatcher": [
 				"$swiftc"
 			],
 			"group": "build",
 			"label": "swift: Build All from tasks.json",
-			"detail": "swift build --build-tests"
+			"detail": "swift build --show-bin-path"
```
[the remark this answers is not in the document]
Comment (award999) [reply assets/test/.vscode/tasks.json:31]: So all I want to do was make sure the task executes and environment succeeds in running a Swift command. Decided didn't need to unnecessarily build something. At least that's my take

**Review by plemarquand — APPROVED** [review]

**Pull request merged**